spec/monitoring/pub_sub_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rack/test'
+require 'spec_helper'
+require 'monitoring/pub_sub'
+require 'monitoring/prometheus'
+
+describe Monitoring::PubSub do
+  include Rack::Test::Methods
```

Comment:
Same comment as above. This is really an integration test. Alternatively, you might want to mock the `ActiveSupport::Notifications` methods we're wrapping in `Monitoring::PubSub`. That'll allow you test individual methods more robustly.